### DISCUSSION TITLE
About Haskell Grammar use

### DISCUSSION THREAD

**Discussion by Boroughbuga** [antlr/grammars-v4#3437] [Q&A]
Hello,

I am new to Antlr and I am trying to make use of the Haskell Language Grammar provided here however I am running into some problems when I am trying to execute the examples provided with the grammar.

For example when I simply run this command

antlr4-parse .\HaskellParser.g4 .\HaskellLexer.g4  module -tree \examples\doAndCase.hs

it gives me this error message;
line 1:10 no viable alternative at input 'doGuessing '

It gives pretty much similar errors for all the examples.
What am I doing wrong?

**Comment by kaby76** [comment]
[antlr4-parse](https://github.com/antlr/antlr4-tools#parsing-using-interpreter) uses the interpreter for the grammar. Unfortunately, the interpreter won't work because it cannot execute [target-specific code in the haskell grammar](https://github.com/antlr/grammars-v4/blob/7ec03688034bfe4d98f81accaa2ee421496f3c47/haskell/HaskellLexer.g4#LL199C21-L199C21). A parser and lexer must be generated and compiled. The haskell grammar has [two ports](https://github.com/antlr/grammars-v4/blob/7ec03688034bfe4d98f81accaa2ee421496f3c47/haskell/desc.xml#L3): CSharp and Java.

```
antlr4 HaskellLexer.g4 HaskellParser.g4
cp Java/*.java .
javac -cp ".../antlr4-4.12.0-complete.jar;." *.java
java -cp ".../antlr4-4.12.0-complete.jar;." org.antlr.v4.gui.TestRig Haskell module foobar.hs -tokens -tree -gui
```
